app/api/controllers/book_api.rb

```diff
@@ -0,0 +1,48 @@
+class BookApi < Grape::API
+  resource :books do
+    desc 'Get all books'
+    get do
+      books = Book.all
+      BookSerializer.new(books).serializable_hash
+    end
+
+    desc 'Create a new book'
+    params do
+      requires :name, type: String
+      requires :release, type: Date
+      optional :description
+    end
+    post do
+      book = Book.create(declared(params).merge(user: current_user))
+      BookSerializer.new(book).serializable_hash
+    end
+
+    desc 'Get a specific book'
+    get ':id' do
+      book = Book.find_by(id: params[:id])
+      error!('Not Found', 404) unless book
+      BookSerializer.new(book).serializable_hash
+    end
+
+    desc 'Update a book'
+    params do
+      optional :name, type: String, desc: 'Book name'
+      optional :release, type: Date, desc: 'Release date'
+      optional :description
+    end
+    put ':id' do
+      book = Book.find_by(id: params[:id])
+      error!('Not Found', 404) unless book
+      book.update(params)
+      BookSerializer.new(book).serializable_hash
+    end
+
+    desc 'Delete a book'
+    delete ':id' do
+      book = Book.find_by(id: params[:id])
+      error!('Not Found', 404) unless book
+      book.destroy
+      { message: 'Book deleted successfully' }
+    end
```

Comment:
Better wrap the routes with `:id` into `route_param :id` block as example below:

```suggestion
    route_param :id do
      desc 'GET /api/v1/books/:id'
      get do
        # ...
      end

      desc 'PUT /api/v1/books/:id'
      params do
        optional :name, type: String, desc: 'Book name'
        optional :release, type: Date, desc: 'Release date'
        optional :description
      end
      put do
        # ...
      end

      desc 'DELETE /api/v1/books/:id'
      delete do
        # ...
      end
    end```

Reply:
Always use this style on each desc: `desc 'GET /api/v1/books/:id'`.

Reply:
Completed

- [ ]  Change Desc
- [ ]  Wrap :id

commit: https://github.com/RobertDataWow/rails-book/pull/23/commits/6a1d59ed9a1227f246d60fefa009fc17e5f78dde